jscomp/others/js_date.ml

```diff
@@ -50,7 +50,7 @@ external utcWithYMDHMS : year:float -> month:float -> date:float -> hours:float
 external now : unit -> float = "" [@@bs.val "Date.now"]
 (** returns the number of milliseconds since Unix epoch *)
 
-external parse : string -> t = "Date" [@@bs.new]
+external parse : string -> float = "Date" [@@bs.val "Date.parse"]
```

Comment:
this is a breaking change, can you add a new function like `parseAsFloat`?, also checkout bs.scope

Reply:
Thanks for the review.

Yes, it is a breaking change, but that's because the existing binding was incorrect, at least if the aim of the bindings in the Js namespace is to mirror/follow the JS API as closely as possible.

IMHO `Js.Date.parse` needs to map to `Date.parse`, not `new Date()`.

Reply:
But such change would break people's working code. Adding `parseAsFloat` and deprecate the original `parse` would make both happy

Reply:
Ok, that sounds like a good solution, I will update my PR accordingly. 👍 (Deprecating the original `parse` is important IMHO as it is very confusing because the signature differs from the JS function `Date.parse`.)

As for `bs.scope`, I know about that one, but the other bindings in this file do not use it either, so I followed suit.